### DISCUSSION TITLE
Can't Train, not sure what I did wrong (Exception training model: ''CheckpointInfo' object has no attribute 'sha256''.)

### DISCUSSION THREAD

**Discussion by A-Merk** [d8ahazard/sd_dreambooth_extension#840] [Q&A]
Does anyone know why I am getting this error? I am very new to all of this so I apologize if the answer is obvious but I would love the help if someone knows of an easy fix.

Error is: "Exception training model: ''CheckpointInfo' object has no attribute 'sha256''."

**Comment by A-Merk** [comment]
Custom model name is Flowerpunkv1
Initializing dreambooth training...
Pre-processing flowerpunkv1: 100%|█████████████████████████████████████████████████████| 33/33 [00:00<00:00, 93.48it/s]
Pre-processing classifiers_0: 0it [00:00, ?it/s]
Concept requires 10 class images per instance image.
Sorting instance images: 100%|██████████████████████████████████████████████████████████| 1/1 [00:00<00:00, 999.83it/s]
We need a total of 330 class images.
Generating 330 class images for training...
Generating class images 0/330::   0%|                                                          | 0/330 [00:00<?, ?it/s]Traceback (most recent call last):
  File "C:\Users\alexm\stable-diffusion-webui\extensions\sd_dreambooth_extension\scripts\dreambooth.py", line 455, in start_training
    result = main(config, use_txt2img=use_txt2img)
  File "C:\Users\alexm\stable-diffusion-webui\extensions\sd_dreambooth_extension\dreambooth\train_dreambooth.py", line 989, in main
    return inner_loop()
  File "C:\Users\alexm\stable-diffusion-webui\extensions\sd_dreambooth_extension\dreambooth\memory.py", line 121, in decorator
    return function(batch_size, grad_size, prof, log_file, *args, **kwargs)
  File "C:\Users\alexm\stable-diffusion-webui\extensions\sd_dreambooth_extension\dreambooth\train_dreambooth.py", line 177, in inner_loop
    count, instance_prompts, class_prompts = generate_classifiers(args, use_txt2img=use_txt2img, accelerator=accelerator, ui = False)
  File "C:\Users\alexm\stable-diffusion-webui\extensions\sd_dreambooth_extension\dreambooth\finetune_utils.py", line 942, in generate_classifiers
    builder = ImageBuilder(args, use_txt2img=use_txt2img, lora_model=args.lora_model_name,
  File "C:\Users\alexm\stable-diffusion-webui\extensions\sd_dreambooth_extension\dreambooth\finetune_utils.py", line 486, in __init__
    if isinstance(new_model_info, sd_models.CheckpointInfo) and new_model_info.sha256 != current_model.sha256:
AttributeError: 'CheckpointInfo' object has no attribute 'sha256'

is the full error.

Any help? I got this error when trying to train both SD1.4 and SD1.5. SD1.4 passes the 256 check when using deforum so I have no idea what's going on, would love some help if anyone knows...

**Comment by dote78** [comment]
Same here. Same error messages. 

Source Checkpoint: F:\StableDiffusion\stable-diffusion-webui\models\Stable-diffusion\v1-5-pruned-emaonly.ckpt
Scheduler: euler-ancestral

Opening with --no-half --xformers --lowvram
Ryzen 3700 with a 3070 GPU

**Reply by dote78** [reply]
Adding more info about the setting:

Checking Dreambooth requirements...
[+] bitsandbytes version 0.35.0 installed.
[+] diffusers version 0.10.2 installed.
[+] transformers version 4.25.1 installed.
[+] xformers version 0.0.14.dev0 installed.
[+] torch version 1.12.1+cu113 installed.
[+] torchvision version 0.13.1+cu113 installed.

Should it be running torch 1.13.1 and xformers 0.0.16rc425? It gives that message upon starting....

Launching Web UI with arguments: --no-half --xformers --lowvram
==============================================================================
You are running torch 1.12.1+cu113.
The program is tested to work with torch 1.13.1.
To reinstall the desired version, run with commandline flag --reinstall-torch.
Beware that this will cause a lot of large files to be downloaded, as well as
there are reports of issues with training tab on the latest version.

Use --skip-version-check commandline argument to disable this check.
==============================================================================
=================================================================================
You are running xformers 0.0.14.dev.
The program is tested to work with xformers 0.0.16rc425.
To reinstall the desired version, run with commandline flag --reinstall-xformers.

Use --skip-version-check commandline argument to disable this check.
=================================================================================

**Comment by dote78** [comment]
Ok, with the latest updates these past days, now it seems to be working properly. I just tried again with the same settings as before and it is now training ok. Seems to have been solved. Thanks
